packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -266,22 +278,40 @@ export class Alarm extends AlarmBase {
                   stat: stat.statistic,
                   unit: stat.unitFilter,
                 },
-                id: entry.id || uniqueMetricId(),
+                id: entry.id,
                 label: conf.renderingProperties?.label,
                 returnData: entry.tag ? undefined : false, // Tag stores "primary" attribute, default is "true"
               };
             },
             withExpression(expr, conf) {
               return {
                 expression: expr.expression,
-                id: entry.id || uniqueMetricId(),
+                id: entry.id,
```

Comment:
why was this removed?

Reply:
Not removed, but [relocated above](https://github.com/aws/aws-cdk/pull/10545/files/b167b86e3a2ab5f261d20a54d71bf75ff98a9d18#diff-305a2ad13e2f1fb31319d17f880546d1R263) so I could apply this in [anomalyDetectionMetricIds](https://github.com/aws/aws-cdk/pull/10545/files/b167b86e3a2ab5f261d20a54d71bf75ff98a9d18#diff-305a2ad13e2f1fb31319d17f880546d1R299)